pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -176,46 +172,37 @@ func runCommand(command string, args ...string) (string, error) {
 }
 
 func getMlnxNicFwData(pciAddress string) (current, next *mlnxNic, err error) {
-	glog.Infof("mellanox-plugin getMlnxNicFwData(): for device %s", pciAddress)
+	glog.Infof("mellanox-plugin getMlnxNicFwData(): device %s", pciAddress)
 	err = nil
 	attrs := []string{TotalVfs, EnableSriov}
-	singlePort, err := isSinglePortNic(pciAddress)
-	if err != nil {
-		glog.Errorf("mellanox-plugin getMlnxNicFwData(): %v", err)
-		return
-	}
 	firstPort := isFirstPort(pciAddress)
 
-	if singlePort {
-		attrs = append(attrs, "LINK_TYPE")
+	if firstPort {
+		attrs = append(attrs, "LINK_TYPE_P1")
 	} else {
-		if firstPort {
-			attrs = append(attrs, "LINK_TYPE_P1")
-		} else {
-			attrs = append(attrs, "LINK_TYPE_P2")
-		}
+		attrs = append(attrs, "LINK_TYPE_P2")
 	}
-	out, err := mstconfigReadData(pciAddress, attrs...)
+	out, err := mstConfigReadData(pciAddress)
 	if err != nil {
 		return
 		glog.Errorf("mellanox-plugin getMlnxNicFwData(): %v", err)
 	}
 	mstCurrentData, mstNextData := parseMstconfigOutput(out, attrs)
-	current, err = mlnxNicFromMap(mstCurrentData, singlePort, firstPort)
+	current, err = mlnxNicFromMap(mstCurrentData, firstPort)
 	if err != nil {
 		glog.Errorf("mellanox-plugin getMlnxNicFwData(): %v", err)
 		return
 	}
-	next, err = mlnxNicFromMap(mstNextData, singlePort, firstPort)
+	next, err = mlnxNicFromMap(mstNextData, firstPort)
 	if err != nil {
 		glog.Errorf("mellanox-plugin getMlnxNicFwData(): %v", err)
 	}
 	return
 }
 
-func mlnxNicFromMap(mstData map[string]string, singlePort, firstPort bool) (*mlnxNic, error) {
-	glog.Infof("mellanox-plugin mlnxNicFromMap() %v, %v, %v", mstData, singlePort, firstPort)
-	fwData := &mlnxNic{singlePort: singlePort, firstPort: firstPort}
+func mlnxNicFromMap(mstData map[string]string, firstPort bool) (*mlnxNic, error) {
+	glog.Infof("mellanox-plugin mlnxNicFromMap() %v, %v", mstData, firstPort)
+	fwData := &mlnxNic{firstPort: firstPort}
```

Comment:
TBH, i dont think mlxNic struct should contain first port attribute.
it should contain the PCI address and a bound method of getPortNum()

Reply:
The port number can only be 0 or 1, there is no need to create the function getPortNumber, instead, I will move isFirtPort function to the struct